runtime/test_artifacts/generator.go

```diff
@@ -6,108 +6,149 @@ import (
+			default:
+				panic(fmt.Sprintf("invalid actor type: %s", actor.ActorType))
+			}
+		}
+	}
+}
+
 func NewDefaultConfigs(privateKeys []string) (cfgs []*configs.Config) {
 	for i, pk := range privateKeys {
+		postgresSchema := "node" + strconv.Itoa(i+1)
 		cfgs = append(cfgs, configs.NewDefaultConfig(
 			configs.WithPK(pk),
-			configs.WithNodeSchema("node"+strconv.Itoa(i+1)),
+			configs.WithNodeSchema(postgresSchema),
 		))
 	}
-	return
+	return cfgs
 }
 
-// REFACTOR: Test artifact generator should reflect the sum of the initial account values to populate the initial pool values
 func NewPools() (pools []*coreTypes.Account) {
 	for _, value := range coreTypes.Pools_value {
 		if value == int32(coreTypes.Pools_POOLS_UNSPECIFIED) {
 			continue
 		}
 
+		// TECHDEBT: Test artifact should reflect the sum of the initial account values
+		// rather than be set to `DefaultAccountAmountString`
 		amount := DefaultAccountAmountString
 		if value == int32(coreTypes.Pools_POOLS_FEE_COLLECTOR) {
-			amount = "0"
+			amount = "0" // fees are empty at genesis
 		}
 
+		poolAddr := hex.EncodeToString(coreTypes.Pools(value).Address())
+
 		pools = append(pools, &coreTypes.Account{
-			Address: hex.EncodeToString(coreTypes.Pools(value).Address()),
+			Address: poolAddr,
 			Amount:  amount,
 		})
 	}
-	return
+	return pools
+}
+
+func newAccountsWithKeys(privateKeys []string) (accounts []*coreTypes.Account) {
+	for _, pk := range privateKeys {
+		pk, _ := crypto.NewPrivateKey(pk)
+		addr := pk.Address().String()
+		accounts = append(accounts, &coreTypes.Account{
+			Address: addr,
+			Amount:  DefaultAccountAmountString,
+		})
+	}
+	return accounts
 }
 
-func NewAccounts(n int, privateKeys ...string) (accounts []*coreTypes.Account) {
-	for i := 0; i < n; i++ {
+//nolint:unused
+func newAccounts(numActors int) (accounts []*coreTypes.Account) {
+	for i := 0; i < numActors; i++ {
 		_, _, addr := keygen.GetInstance().Next()
-		if privateKeys != nil {
-			pk, _ := crypto.NewPrivateKey(privateKeys[i])
-			addr = pk.Address().String()
-		}
 		accounts = append(accounts, &coreTypes.Account{
 			Address: addr,
 			Amount:  DefaultAccountAmountString,
 		})
 	}
-	return
+	return accounts
 }
 
-// TODO: The current implementation of NewActors  will have overlapping `ServiceUrl` for different
-//
-//	types of actors which needs to be fixed.
-func NewActors(actorType coreTypes.ActorType, n int) (actors []*coreTypes.Actor, privateKeys []string) {
-	for i := 0; i < n; i++ {
+// TECHDEBT: Current implementation of `NewActors` will result in non-unique ServiceURLs if called
+// more than once.
+func NewActors(actorType coreTypes.ActorType, numActors int, chains []string) (actors []*coreTypes.Actor, privateKeys []string) {
+	// If the actor type is a validator, the chains must be nil since they are chain agnostic
+	if actorType == coreTypes.ActorType_ACTOR_TYPE_VAL {
+		logger.Global.Warn().Msgf("validator actors should not have chains but a list was provided: %v", chains)
```

Comment:
I don't think `testutils` is appropriate because we're using it outside of tests as well.

Alternatively, I don't think `shared/*_utils.go` is also appropriate because it's very logging specific.

I've moved it to `logger/utils.go`. Lmk what you think

Reply:
Perfect! :ok_hand: